MiniSim/Menu.swift

```diff
@@ -121,103 +155,125 @@ class Menu: NSMenu {
+            let menuItems = devices.map { createMenuItem(for: $0) }
+            self.updateSection(with: menuItems, section: section)
         }
     }
     
-    private func populateAndroidSubMenu(booted: Bool) -> NSMenu {
-        let subMenu = NSMenu()
-        for item in AndroidSubMenuItem.allCases {
-            if item == AndroidSubMenuItem.customCommand {
-                continue
-            }
-            
-            let menuItem = item.menuItem
-            menuItem.target = self
-            menuItem.action = #selector(androidSubMenuClick)
-            if item.needBootedDevice && !booted {
-                continue
-            }
-            if item.bootsDevice && booted {
-                continue
-            }
-            subMenu.addItem(menuItem)
+    var sections: [DeviceListSection] {
+        var sections: [DeviceListSection] = []
+        if UserDefaults.standard.enableAndroidEmulators {
+            sections.append(.android)
         }
-        
-        for item in DeviceService.getCustomCommands(platform: .android) {
-            let menuItem = AndroidSubMenuItem.customCommand.menuItem
-            menuItem.target = self
-            menuItem.action = #selector(androidSubMenuClick)
-            if item.needBootedDevice && !booted {
-                continue
-            }
-            if item.bootsDevice ?? false && booted {
-                continue
-            }
-            menuItem.image = NSImage(systemSymbolName: item.icon, accessibilityDescription: item.name)
-            menuItem.title = item.name
-            subMenu.addItem(menuItem)
+        if UserDefaults.standard.enableiOSSimulators {
+            sections.append(.iOS)
         }
-        return subMenu
+        return sections
     }
     
-    private func populateIOSSubMenu(booted: Bool) -> NSMenu {
-        let subMenu = NSMenu()
-        for item in IOSSubMenuItem.allCases {
-            if item == IOSSubMenuItem.customCommand {
+    private func filter(devices: [Device], for section: DeviceListSection) -> [Device] {
+        let platform: Platform = section == .iOS ? .ios : .android
+        return devices.filter { $0.platform == platform }
+    }
+    
+    private func updateSection(with items: [NSMenuItem], section: DeviceListSection) {
+        guard let header = self.items.first(where: { $0.tag == section.rawValue }),
+              let startIndex = self.items.firstIndex(of: header) else {
+            return
+        }
+        
+        for (index, menuItem) in items.enumerated() {
+            if let itemIndex = self.items.firstIndex(where: { $0.title == menuItem.title }) {
+                self.replaceMenuItem(at: itemIndex, with: menuItem)
                 continue
             }
-            let menuItem = item.menuItem
-            menuItem.target = self
-            menuItem.action = #selector(IOSSubMenuClick)
-            subMenu.addItem(menuItem)
+            self.safeInsertItem(menuItem, at: startIndex + index + 1)
```

Comment:
This logic needs to be aligned to account for the separator. 

You can change it like so (If you won't find better solution):

```swift
self.safeInsertItem(menuItem, at: section == .iOS ? startIndex + 1 : startIndex + index + 1)
```